### PR TITLE
Fix doctrine/data-fixtures PR#334

### PR DESCRIPTION
**Pull request by stephen-lewis**
PR #334  broke existing code by adding quotes in the middle of table names that used schemas in SQLite - as reported in #338 . 

I've reworked it into a cleaner function that mimics SchemaTool in doctrine/orm. Tests included and pass locally; I've also setup and run a simple test project locally using schema and non-schema entities without issue.

### CODE REVIEW

**Comment by greg0ire** [comment]
@stephen-lewis can you please run [this script](https://gist.github.com/greg0ire/a404831add1247d2bc20fa11107c5d5e) from your fork? It should produce introduce for people who need to test.

File: lib/Doctrine/Common/DataFixtures/Purger/ORMPurger.php [hunk 1]
```diff
@@ -274,4 +269,17 @@ private function getJoinTableName($assoc, $class, $platform)
 
         return $this->em->getConfiguration()->getQuoteStrategy()->getJoinTableName($assoc, $class, $platform);
     }
+
+    /**
+     * @param string              $tableName
+     * @param AbstractPlatform    $platform
+     *
+     * @return string
+     */
```
Comment (greg0ire) [inline lib/Doctrine/Common/DataFixtures/Purger/ORMPurger.php]: Please remove this entire docblock, it's redundant.

**Comment by stephen-lewis** [comment]
> @stephen-lewis can you please run [this script](https://gist.github.com/greg0ire/a404831add1247d2bc20fa11107c5d5e) from your fork? It should produce introduce for people who need to test.

Sure; output below:

# How can I test this?
```shell
composer config repositories.stephen-lewis vcs https://github.com/stephen-lewis/data-fixtures
 "dev-bugfix/fix_errors_from_PR334 as 1.4.1"
```

File: lib/Doctrine/Common/DataFixtures/Purger/ORMPurger.php (hunk 1 above)
Comment (stephen-lewis) [reply lib/Doctrine/Common/DataFixtures/Purger/ORMPurger.php]: Type hints over phpdoc, got it; will tweak now

**Comment by greg0ire** [comment]
@jgxvx please use the instructions above to test this on your project and report back :pray: You might have to use the `no-api` key, but I'm not 100% sure about that, please see a usage example here: https://github.com/composer/composer/issues/5433

**Comment by greg0ire** [comment]
Please kindly squash your commits together. If you don't, we'll try to remember to do it for you but it's best if you save us this trouble.

How to do that?

1. `git rebase -i origin/1.4.x`, assuming `origin` is a git remote that points to this repository, and not your fork. If you're not sure what your remotes are, run `git remote -vvv`, there should be your fork and the holy/reference/base/origin/whatever-you-call-it repository.
2. A window will show up with many lines, replace `pick` with `fixup` on every line but the first one
3. Close your editor, git should do its magic, and you should end up with one commit
4. Use `git push --force` to overwrite what you already push. Don't forget the `--force` option otherwise git will try to merge both things together.

**Comment by jgxvx** [comment]
@greg0ire @stephen-lewis  Glad to confirm that the fix works!

**Comment by greg0ire** [comment]
@stephen-lewis please improve your commit message according to [the contributing guide](https://www.doctrine-project.org/contribute/index.html#working-on-topic-branches).

TL;DR  say what was broken and how your changes improve the situation

**Comment by stephen-lewis** [comment]
@greg0ire commit message should now be updated :smile:
